Update Go version & Dependencies

EyalDelarea opened the pull request:
- [x] All [tests](https://github.com/jfrog/jfrog-cli/CONTRIBUTING.md#tests) have passed. If this feature is not already covered by the tests, new tests have been added.
- [x] The pull request is targeting the `dev` branch.
- [x] The code has been validated to compile successfully by running `go vet ./...`.
- [x] The code has been formatted properly using `go fmt ./...`.

---

sverdlov93 reviewed (commented):
go 1.23.2 > go 1.23.3

EyalDelarea commented:
I've updated the Jenkinsfile and the dockerfile with the new go version
https://github.com/jfrog/jfrog-ecosystem-integration-env/pull/49 - intergraion-env PR
And also updated our Jenkins tool to go 1.23.3
@sverdlov93
